### PR TITLE
@Indexed enhancements

### PR DESCRIPTION
**Pull request by Aloren**
- Move `@Indexed` annotation to high-level annotation package
- `@Indexed`: support index collection type
- `@Indexed` default index name will be: set_field_type_collectionType

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/query/model/Index.java [hunk 1]
```diff
@@ -31,41 +29,18 @@
  * @author Peter Milne
  * @author Anastasiia Smirnova
  */
+@Value
+@Builder
+@RequiredArgsConstructor
```
Comment (tdanylchuk) [inline src/main/java/org/springframework/data/aerospike/query/model/Index.java:34]: why do you need constructor + builder?

File: src/main/java/org/springframework/data/aerospike/index/ReactiveAerospikePersistenceEntityIndexCreator.java [hunk 2]
```diff
@@ -49,12 +49,12 @@ protected void installIndexes(Set<AerospikeIndexDefinition> indexes) {
...
 
-    private Mono<Void> installIndex(AerospikeIndexDefinition definition) {
-        log.debug("Installing aerospike index: {}...", definition);
-        return template.createIndex(definition.getEntityClass(), definition.getName(), definition.getFieldName(), definition.getType())
-                .doOnSuccess(__ -> log.info("Installed aerospike index: {} successfully.", definition))
-                .onErrorResume(IndexAlreadyExistsException.class, e -> onIndexAlreadyExists(e, definition))
-                .doOnError(throwable -> log.error("Failed to install aerospike index: " + definition, throwable));
+    private Mono<Void> installIndex(AerospikeIndexDefinition index) {
+        log.debug("Installing aerospike index: {}...", index);
+        return template.createIndex(index.getEntityClass(), index.getName(), index.getFieldName(), index.getType(), index.getCollectionType())
```
Comment (tdanylchuk) [inline src/main/java/org/springframework/data/aerospike/index/ReactiveAerospikePersistenceEntityIndexCreator.java:54]: btw maybe good to have an overloaded method which accepts ```AerospikeIndexDefinition```

File: src/main/java/org/springframework/data/aerospike/index/AerospikeIndexResolver.java [hunk 3]
```diff
@@ -41,13 +42,20 @@ private AerospikeIndexDefinition convertToIndex(BasicAerospikePersistentEntity<?
...
                 .fieldName(property.getFieldName())
                 .name(indexName)
                 .type(annotation.type())
+                .collectionType(annotation.collectionType())
                 .build();
     }
+
+    private String getIndexName(BasicAerospikePersistentEntity<?> entity,
+                                AerospikePersistentProperty property, Indexed annotation) {
+        return String.join("_",
```
Comment (tdanylchuk) [inline src/main/java/org/springframework/data/aerospike/index/AerospikeIndexResolver.java:58]: what is max length of the index allowed on aerospike? ;D

**Review by tdanylchuk — APPROVED** [review]

File: src/main/java/org/springframework/data/aerospike/index/AerospikeIndexResolver.java (hunk 3 above)
Comment (Aloren) [reply src/main/java/org/springframework/data/aerospike/index/AerospikeIndexResolver.java:58]: 255 bytes, should be enough 😎

**Pull request merged**